lib/transaction.rb

```diff
@@ -0,0 +1,12 @@
+class Transaction
+
+  attr_reader :list
+
+  def initialize
+    @list = []
+  end
+
+  def store_transaction(date, amount, balance)
+    self.list.push({date: date, amount: amount, balance: balance})
```

Comment:
OO code tends to avoid using plain data structures.  Instead, it encapsulates it inside a class.